### PR TITLE
EMP nerfs

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

The time that EMPs disable guns and stun batons for has been updated:
Light EMP: 0 to 10 seconds
Heavy EMP: 0 to 20 seconds

## Why It's Good For The Game

A single EMP can make any member of security unable to fight back for a very long time. This makes EMP temporarilly disable them during a fight, so that you can get a short upper hand however the other person can fight back after at most 20 seconds rather than 2 minutes.
A lot of complaints that this required the use of shotgun meta, so this should make it so you can still fight against someone with EMPs with the user of regular weapons.

## Testing Photographs and Procedure

https://user-images.githubusercontent.com/26465327/172069399-f7f1091a-2ece-4921-8462-a07448306c20.mp4

## Changelog
:cl:
balance: Light EMPs will disable energy weapons and batons for 0-10 seconds as opposed to 30-60 seconds.
balance: Heavy EMPs will disable energy weapons and batons for 0-20 seconds as opposed to 60-120 seconds.
/:cl:

### CODE REVIEW

**Comment by AnCopper** [comment]
> A single EMP can make any member of security unable to fight back for a very long time. 

This is the EXACT reason why EMP's didn't affect stun batons as much, until you changed it. This is literally a problem that you yourself have created with #6135

**Comment by PowerfulBacon** [comment]
Well one of the most powerful weapons in the game should have some pretty powerful downsides too. This is overall a better solution than the original.

**Comment by KennyBadrige** [comment]
speedmerge

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -201,7 +201,7 @@
 	if (!(. & EMP_PROTECT_SELF) && !(obj_flags & OBJ_EMPED))
 		obj_flags |= OBJ_EMPED
 		update_icon()
-		addtimer(CALLBACK(src, .proc/emp_reset), rand(1200 / severity, 600 / severity))
+		addtimer(CALLBACK(src, .proc/emp_reset), rand(0, 200 / severity))
```
Comment (Archanial) [inline code/game/objects/items/stunbaton.dm]: Possible runtime

File: code/modules/projectiles/guns/energy.dm [hunk 2]
```diff
@@ -4,38 +4,38 @@
...
-	var/dead_cell = FALSE 
+	var/dead_cell = FALSE
 
 /obj/item/gun/energy/emp_act(severity)
 	. = ..()
 	if(!(. & EMP_PROTECT_CONTENTS))
 		obj_flags |= OBJ_EMPED
 		update_icon()
-		addtimer(CALLBACK(src, .proc/emp_reset), rand(600 / severity, 300 / severity))
+		addtimer(CALLBACK(src, .proc/emp_reset), rand(0, 200 / severity))
```
Comment (Archanial) [inline code/modules/projectiles/guns/energy.dm]: Possible runtine

**Review by Archanial — DISMISSED** [review]
You can't call timers with 0 time

File: code/modules/projectiles/guns/energy.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/projectiles/guns/energy.dm]: How?

Comment (PowerfulBacon) [reply code/modules/projectiles/guns/energy.dm]: Oh 0 length timers

File: code/game/objects/items/stunbaton.dm (hunk 1 above)
Comment (PowerfulBacon) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
		addtimer(CALLBACK(src, .proc/emp_reset), rand(1, 200 / severity))
```

File: code/modules/projectiles/guns/energy.dm (hunk 2 above)
Comment (PowerfulBacon) [inline code/modules/projectiles/guns/energy.dm]:
```suggestion
		addtimer(CALLBACK(src, .proc/emp_reset), rand(1, 200 / severity))
```

**Comment by Kratairo** [comment]
Oh no sec main pr 😔

**Comment by ghost** [comment]
I'm personally a big fan of making EMPs less common against security, it's already annoying enough when you try play IPC security and the frequent EMPing just kills you.

**Comment by Isy232** [comment]
I agree the previous was too much, but I think 0-10 or 0-20 is too little.

Perhaps 10-20 and 20-30 would be superior

Remember that EMP's used to drain energy guns almost entirely making them useless until recharged. If a heavy EMP has the chance to disable an energy gun for only a single second, most antagonists will likely not wish to use them based on their inconsistency.

There's already a lot of things that are inconsistent with antagonists, perhaps they shouldn't have to worry about their EMP not working at all.

**Comment by FortSumter** [comment]
Or just give security glocks like NSV and like... you know...

Real police....

**Comment by PowerfulBacon** [comment]
> Remember that EMP's used to drain energy guns almost entirely making them useless until recharged. If a heavy EMP has the chance to disable an energy gun for only a single second, most antagonists will likely not wish to use them based on their inconsistency.

Considering how powerful EMPs are, a nerf that will make people less likely to use them is a good thing in my opinion.
You are guaranteed to disable their weapons for at least a single hit, allowing you to get the upper hand, however an EMP won't immediately decide the outcome of the battle. If someone is robust and has a lot of weapons, one of their weapons is likely to reactivate very quickly, allowing them to switch to that weapon and use it (even if its not the first one they would have chosen for the situation).

**Review by Rukofamicom — APPROVED** [review]
Code checks out, only two lines actually changed with the rest of the "changed" lines just being cleanup of empty space. 
This review means nothing, but I feel bad about it sitting here for so long and want to see it merged :P

Regarding the balance: Extremely hard counters bad. Softer counters good.

**Comment by PowerfulBacon** [comment]
What a great PR, and very good balance changes. I put my official approval on this PR :+1:

**Review by ike709 — APPROVED** [review]

**Pull request merged**